src/water_cmd.cpp

```diff
@@ -348,8 +348,10 @@ static CommandCost DoBuildLock(TileIndex tile, DiagDirection dir, DoCommandFlags
 	}
 	WaterClass wc_upper = IsWaterTile(tile + delta) ? GetWaterClass(tile + delta) : WATER_CLASS_CANAL;
 
-	if (IsBridgeAbove(tile) || IsBridgeAbove(tile - delta) || IsBridgeAbove(tile + delta)) {
-		return CommandCost(STR_ERROR_MUST_DEMOLISH_BRIDGE_FIRST);
+	for (TileIndex t : {tile, tile - delta, tile + delta}) {
+		if (IsBridgeAbove(t) && GetBridgeHeight(GetSouthernBridgeEnd(t)) < GetTileMaxZ(t) + 2) {
+			return CommandCost(STR_ERROR_MUST_DEMOLISH_BRIDGE_FIRST);
```

Comment:
Oh, and should this be an error saying the bridge is too low, as with stations/docks? Instead of this generic error?